crates/bevy_ecs/src/query/filter.rs

```diff
@@ -443,21 +443,17 @@ macro_rules! impl_or_query_filter {
             fn update_component_access(state: &Self::State, access: &mut FilteredAccess<ComponentId>) {
                 let ($($filter,)*) = state;
 
-                let mut _new_access = access.clone();
-                let mut _not_first = false;
+                let mut _new_access = FilteredAccess::empty();
+
                 $(
-                    if _not_first {
-                        let mut intermediate = access.clone();
-                        $filter::update_component_access($filter, &mut intermediate);
-                        _new_access.append_or(&intermediate);
-                        _new_access.extend_access(&intermediate);
-                    } else {
-                        $filter::update_component_access($filter, &mut _new_access);
-                        _new_access.required = access.required.clone();
-                        _not_first = true;
-                    }
+                    let mut intermediate = access.clone();
+                    $filter::update_component_access($filter, &mut intermediate);
+                    _new_access.append_or(&intermediate);
+                    _new_access.extend_access(&intermediate);
                 )*
 
+                _new_access.required = access.required.clone();
```

Comment:
I think this line can be removed as well; they were needed just because we had to copy the access to seed the accumulation

Reply:
The seed is still an empty `FilteredAccess`, just using that will lose informations about all the required components.

I can however avoid the `.clone()` here and instead use `std::mem::take`.